terraformignore.go

```diff
@@ -74,22 +74,28 @@ func readRules(input io.Reader) []rule {
 	return rules
 }
 
-func matchIgnoreRule(path string, rules []rule) bool {
-	matched := false
+func matchIgnoreRule(path string, rules []rule) (bool, *rule) {
+	var matched *rule = nil
 	path = filepath.FromSlash(path)
-	for _, rule := range rules {
+	for idx := range rules {
+		rule := &rules[idx]
```

Comment:
i don't know if there's an idiomatic go way to express this, but this is a pretty significant performance improvement